### PR TITLE
remove requires from preload

### PR DESCRIPTION
**Pull request by bridiver**
includes fix mentioned in #581

### CODE REVIEW

File: app/content/webviewPreload.js [hunk 1]
```diff
@@ -2,255 +2,377 @@
...
+    DOM_VK_SLASH: 191,
+    DOM_VK_BACK_QUOTE: 192,
+    DOM_VK_OPEN_BRACKET: 219,
+    DOM_VK_BACK_SLASH: 220,
+    DOM_VK_CLOSE_BRACKET: 221,
+    DOM_VK_QUOTE: 222,
+    DOM_VK_META: 224
   }
 }
-addEventListener('message', ipc.processMessage.bind(ipc)) // eslint-disable-line
```
Comment (bridiver) [inline app/content/webviewPreload.js]: Most of this didn't change, I just wrapped it in a function call to isolate it from the rest of the page. I'll mark the actual changes below

File: app/content/webviewPreload.js [hunk 2]
```diff
@@ -2,255 +2,377 @@
...
+    on: function (name, cb) {
+      this.events[name] = cb
+    },
+    send: function () {
+      var args = Array.prototype.slice.call(arguments)
+      this.embedder && this.embedder.postMessage(args, 'file://')
+    }
   }
-}
+  window.addEventListener('message', ipc.processMessage.bind(ipc))
```
Comment (bridiver) [inline app/content/webviewPreload.js]: window. added from previous comment

File: app/content/webviewPreload.js [hunk 3]
```diff
@@ -2,255 +2,377 @@
...
+        if (node.parentNode) {
+          ensureNodeVisible(node.parentNode.parentNode)
+        }
       }
     }
   }
-}
 
-// Fires when the browser has ad replacement information to give
-ipc.on(messages.SET_AD_DIV_CANDIDATES, function (e, adDivCandidates, placeholderUrl) {
```
Comment (bridiver) [inline app/content/webviewPreload.js]: messages replaced with message text

File: app/content/webviewPreload.js [hunk 4]
```diff
@@ -2,255 +2,377 @@
...
+
+  document.onkeydown = (e) => {
+    switch (e.keyCode) {
+      case KeyEvent.DOM_VK_ESCAPE:
+        e.preventDefault()
+        ipc.send('stop-load')
+        break
+      case KeyEvent.DOM_VK_BACK_SPACE:
+        if (!isEditable(document.activeElement)) {
+          e.shiftKey ? window.history.forward() : window.history.back()
```
Comment (bridiver) [inline app/content/webviewPreload.js]: window. added from previous comment

File: app/content/webviewPreload.js [hunk 5]
```diff
@@ -2,255 +2,377 @@
...
+      isContentEditable: e.target.isContentEditable,
+      hasSelection: hasSelection(e.target)
+    }
+    ipc.send('context-menu-opened', nodeProps)
+    e.preventDefault()
+  }, false)
+
+  document.onkeydown = (e) => {
+    switch (e.keyCode) {
+      case KeyEvent.DOM_VK_ESCAPE:
```
Comment (bridiver) [inline app/content/webviewPreload.js]: use codes from KeyEvent

**Comment by bbondy** [comment]
lgtm let's roll with it.

File: app/content/webviewPreload.js [hunk 6]
```diff
@@ -2,255 +2,377 @@
...
-  on: function (name, cb) {
-    this.events[name] = cb
-  },
-  send: function () {
-    var args = Array.prototype.slice.call(arguments)
-    this.embedder && this.embedder.postMessage(args, 'file://')
+// inject missing DOM Level 3 KeyEvent
+// https://www.w3.org/TR/2001/WD-DOM-Level-3-Events-20010410/DOM3-Events.html#events-Events-KeyEvent
+if (typeof KeyEvent === 'undefined') {
+  var KeyEvent = {
```
Comment (diracdeltas) [inline app/content/webviewPreload.js]: this should also go in the closure to hide it from the rest of the page?

Comment (bridiver) [inline app/content/webviewPreload.js]: I intentionally left it out. It's part of DOM Level 3 and exists in Firefox, but Chrome didn't implement it so we might as well make it available

**Pull request merged**